### ISSUE TITLE
not finding iio.h

### ISSUE THREAD

**Issue by psiqueira** [tfcollins/homebrew-formulae#12]
Hi Travis,

 thank you for making this formula for installing iio-oscilloscope available.

  unfortunately, I am having trouble running the formula.  I am running OSX 11.3.1 and think that everything is current.  Below is the output from homebrew that says there is an issue with the installation.  In this case it appears to be in finding iio.h.

 do you have any thoughts on how I can troubleshoot the issue?

Thank you, Paul

```
[taiga 734 ] analog_devices : brew install --HEAD tfcollins/homebrew-formulae/iio-oscilloscope
==> Installing iio-oscilloscope from tfcollins/formulae
==> Downloading https://github.com/analogdevicesinc/libad9361-iio/archive/v0.2.tar.gz
Already downloaded: /Users/siqueira/Library/Caches/Homebrew/downloads/21df4020b46564678039f8624df307b6a0496b2f500ad8dc5e08e5d4fd8cf00a--libad9361-iio-0.2.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/szip/manifests/2.1.1_1
Already downloaded: /Users/siqueira/Library/Caches/Homebrew/downloads/9f9b3db3fcab3fb68882ccc2038a29d701f498977476b20f27c1c7cc74ec3d73--szip-2.1.1_1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/szip/blobs/sha256:1779ec8c3312993ef7e22679df6bbcd3adce9db28d3ad98adb54650c018ed294
Already downloaded: /Users/siqueira/Library/Caches/Homebrew/downloads/c0eb9d7b93db68febbb605908d87d63562093643c8b72d646ec06aff0656ea19--szip--2.1.1_1.big_sur.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/hdf5/manifests/1.12.0_4
Already downloaded: /Users/siqueira/Library/Caches/Homebrew/downloads/8035852e2011c5acfc59a0508c54c616484e640f87cf19e8ee51994a4152b1f3--hdf5-1.12.0_4.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/hdf5/blobs/sha256:3302a30ba82ab91f4027f61ce82c867edbbcbe57dd4ce90ad56b91ecec826f82
Already downloaded: /Users/siqueira/Library/Caches/Homebrew/downloads/34fcf95533a9b266641054f91d5dd1d320f387b87903bd495cef3b7866aef019--hdf5--1.12.0_4.big_sur.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/libmatio/manifests/1.5.21
Already downloaded: /Users/siqueira/Library/Caches/Homebrew/downloads/6c5cdbc822d33c904108beda6707f3cdad64eceb8b9d1fe669883e514d62e5b2--libmatio-1.5.21.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/libmatio/blobs/sha256:98a891d294f382a85d61c8d57381e02d1ecb0139af46c08335c8f2eaf5727d7d
Already downloaded: /Users/siqueira/Library/Caches/Homebrew/downloads/d31d81865ca540ab95de8f371f694d8c7ed46bbb0ee7fbf7ad51610208b26f2a--libmatio--1.5.21.big_sur.bottle.tar.gz
==> Cloning https://github.com/analogdevicesinc/iio-oscilloscope.git
Updating /Users/siqueira/Library/Caches/Homebrew/iio-oscilloscope--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at f500bb8 plugins/fmcomms11: Add support for the moved NCO frequency channel
==> Installing dependencies for tfcollins/formulae/iio-oscilloscope: libad9361-iio, szip, hdf5 and libmatio
==> Installing tfcollins/formulae/iio-oscilloscope dependency: libad9361-iio
==> cmake .. -DOSX_PACKAGE=OFF
==> make
Last 15 lines from /Users/siqueira/Library/Logs/Homebrew/libad9361-iio/02.make:
/tmp/libad9361-iio-20210524-5420-sl37m0/libad9361-iio-0.2/ad9361_multichip_sync.c:18:10: fatal error: 'iio.h' file not found
#include <iio.h>
         ^~~~~~~
1 error generated.
1 error generated.
make[2]: *** [CMakeFiles/ad9361.dir/ad9361_multichip_sync.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/ad9361.dir/ad9361_baseband_auto_rate.c.o] Error 1
/tmp/libad9361-iio-20210524-5420-sl37m0/libad9361-iio-0.2/ad9361_design_taps.c:18:10: fatal error: 'iio.h' file not found
#include <iio.h>
         ^~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/ad9361.dir/ad9361_design_taps.c.o] Error 1
make[1]: *** [CMakeFiles/ad9361.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by tfcollins** [comment]
Please make sure libiio is installed first. Assuming you have to tap (`brew tap tfcollins/homebrew-formulae`):
`brew install libiio`

-Travis

**Comment by tfcollins** [comment]
Has your issue been solved?

-Travis

**Comment by tfcollins** [comment]
Closing for now. Reopen if it is still a problem.

-Travis

**Comment by psiqueira** [comment]
Hi Travis,

 thank you for checking!  No, I am still having trouble.  I am not sure what the deal is.

 That is, I had installed from the .pkgs the library for libiio.  I had mentioned this in my previous post, but I did notice that the initial set of errors was coming because it could not find iio.h and iio/iio.h, and was looking for it in the directory “/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers”

to fix that, I took the one that was in /opt/local/include and put it in the Frameworks directory from above.  And while that did fix the problem in finding iio.h, another error popped up:

HEAD is now at f500bb8 plugins/fmcomms11: Add support for the moved NCO frequency channel
==> cmake ..
==> make install
Last 15 lines from /Users/siqueira/Library/Logs/Homebrew/iio-oscilloscope/02.make:
/usr/local/Cellar/gtkdatabox/1.0.0/include/gtkdatabox_markers.h:149:10: error: unknown type name 'GdkRGBA'
                                            GdkRGBA * color, guint size,
                                            ^
/usr/local/Cellar/gtkdatabox/1.0.0/include/gtkdatabox_markers.h:164:4: error: unknown type name 'GdkRGBA'
                        GdkRGBA * color, guint size,
                        ^
/usr/local/Cellar/gtkdatabox/1.0.0/include/gtkdatabox_markers.h:164:4: error: unknown type name 'GdkRGBA'
                        GdkRGBA * color, guint size,
                        ^
12 errors generated.
make[2]: *** [CMakeFiles/osc.dir/osc.c.o] Error 1
12 errors generated.
make[2]: *** [CMakeFiles/osc.dir/oscplot.c.o] Error 1
make[1]: *** [CMakeFiles/osc.dir/all] Error 2
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/tfcollins/homebrew-formulae/issues

In short, I am still having problems.  I am not sure what it is.  If you can think of something obvious that I might be doing wrong, that would be great.  If not, I will keep troubleshooting!!

Paul



On Jun 1, 2021, at 6:15 PM, Travis F. Collins ***@***.******@***.***>> wrote:



Has your issue been solved?

-Travis
